### PR TITLE
Ariasanovsky/issue37

### PR DESCRIPTION
**Pull request by ariasanovsky**
resolvd the issue by separating the Mean Max Gain model and the Action Logits model. This allowed us to adjust learning raetes and notice significant improvements in training data.

However, we uncovered a significant problem with the root selection. For some reason, the next root selected does not correspond to the argmin. This is great news, because it is a straightforward bookkeeping error to fix.

### CODE REVIEW

**Comment by ariasanovsky** [comment]
See Issue #39
